### PR TITLE
feat: add `sync` to synchonize all result buffers

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: src/Compiler.jl [hunk 1]
```diff
@@ -691,7 +691,7 @@ function compile_xla(f, args; client=nothing, optimize=true)
     end
 end
 
-function compile(f, args; client=nothing, optimize=true)
+function compile(f, args; client=nothing, optimize=true, block_until_ready=false)
```
Comment (wsmoses) [inline src/Compiler.jl:694]: Can this also be passed to the macro?

also we should use a better name like async=fals or sync=true

Comment (avik-pal) [reply src/Compiler.jl]: done

Comment (avik-pal) [reply src/Compiler.jl]: oops broke something

**Comment by mofeing** [comment]
I'm ok with this changes, but I don't think `XLA.synced_buffer` is working correctly (at least in macOS).

**Review by wsmoses — APPROVED** [review]

**Pull request merged**